menu_pager.module

```diff
@@ -184,6 +184,9 @@ function theme_menu_pager_next(array $variables) {
  *   An array with 'previous' and 'next' links, if found.
  */
 function menu_pager_get_navigation($menu_link, $restrict_to_parent = FALSE) {
+  if (!isset($menu_link['menu_name'])) {
+    return array();
+  }
   $menu_name = $menu_link['menu_name'];
```

Comment:
@bugfolder Would this work too? If so, I think that'd be better...

```suggestion
  $menu_name = isset($menu_link['menu_name']) ? $menu_link['menu_name'] : array();
```

Reply:
Seems like all subsequent code is expecting `$menu_name` to be a string. If you make it an empty array, then the call to `menu_tree_all_data($menu_name)` is going to be attempting things like this

```
$cid = 'links:' . $menu_name . ':all:' . $mlid . ':' . $langcode . ':' . (int) $max_depth;
 ```

and I think unhappiness will ensue.

Reply:
Hmm, right you are. I think there's an 'array' in there that should be 'string' that tripped me up. How's this instead then?

```suggestion
  $menu_name = isset($menu_link['menu_name']) ? $menu_link['menu_name'] : '';
```

Reply:
That still seems kind of awkward, because, for example, the function `menu_pager_get_navigation()` will now go through the whole process of attempting to build `$navigation` (including calling `menu_tree_all_data(''), passing the empty string), only to end up returning an empty array. Seems cleaner to recognize at the beginning that if there's no `menu_name`, we should just bail out of the function and save some execution time.